### PR TITLE
Filter helper

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

File: src/component/component_system_manager.cc [hunk 1]
```diff
@@ -14,6 +14,19 @@ extern frame_info frame_info;
...
+static bool
+filter_matches_message(comms_msg const &msg, wire_filter_ptr const &filter) {
+    auto &cwire_man = component_system_man.managers.wire_comms_component_man;
+    auto sender = cwire_man.get_instance_data(msg.originator);
+
+    /* If we have no filter defined, we match everything. */
+    if (!filter.wrapped)
+        return true;
+
+    /* We have a filter. Sender must have a filter, and it must match. */
```
Comment (RobotCaleb) [inline src/component/component_system_manager.cc]: sender must have a label?

Comment (chrisforbes) [reply src/component/component_system_manager.cc]: If filter requires a label, then yes.

Comment (chrisforbes) [reply src/component/component_system_manager.cc]: The first case  above catches the case where sender doesnt have a label and filter doesnt require anything

Comment (chrisforbes) [reply src/component/component_system_manager.cc]: OH -- are you complaining about my slightly wrong comment, rather than the logic?

Comment (RobotCaleb) [reply src/component/component_system_manager.cc]: The comment says filter. But sender doesn't require a filter and you check for a label.

Comment (RobotCaleb) [reply src/component/component_system_manager.cc]: Yes

**Pull request merged**